extensions/ffmpeg/src/main/java/com/google/android/exoplayer2/ext/ffmpeg/FfmpegDecoder.java

```diff
@@ -87,7 +87,11 @@ public FfmpegDecoderException decode(DecoderInputBuffer inputBuffer,
     }
     if (!hasOutputFormat) {
       channelCount = ffmpegGetChannelCount(nativeContext);
-      sampleRate = ffmpegGetSampleRate(nativeContext);
+      if ("alac".equals(codecName)) {
```

Comment:
The ALAC decoder in FFmpeg does not set `sample_rate` on the native context the same way the Vorbis, Opus, and FLAC decoders do. Because of this, the sample rate needs to be obtained from the last 4 bytes in the ALAC cookie somehow. FFmpeg just [skips them](https://github.com/FFmpeg/FFmpeg/blob/master/libavcodec/alac.c#L541).